libs/algorithms/CMakeLists.txt

```diff
@@ -0,0 +1,125 @@
+# Copyright (c) 2019 The STE||AR-Group
+#
+# Distributed under the Boost Software License, Version 1.0. (See accompanying
+# file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+
+cmake_minimum_required(VERSION 3.3.2 FATAL_ERROR)
+
+set(algorithms_headers
+  hpx/parallel/util/scan_partitioner.hpp
+  hpx/parallel/util/partitioner.hpp
+  hpx/parallel/util/loop.hpp
+  hpx/parallel/util/foreach_partitioner.hpp
+  hpx/parallel/util/detail/chunk_size.hpp
+  hpx/parallel/algorithms/sort.hpp
+  hpx/parallel/algorithms/partition.hpp
+  hpx/parallel/algorithms/minmax.hpp
+  hpx/parallel/algorithms/merge.hpp
+  hpx/parallel/algorithms/for_loop_reduction.hpp
+  hpx/parallel/algorithms/for_loop.hpp
+  hpx/parallel/algorithms/detail/set_operation.hpp
+  hpx/parallel/algorithms/detail/predicates.hpp
+  hpx/parallel/algorithms/copy.hpp
+  hpx/parallel/container_algorithms/unique.hpp
+  hpx/parallel/container_algorithms/transform.hpp
+  hpx/parallel/container_algorithms/sort.hpp
+  hpx/parallel/container_algorithms/search.hpp
+  hpx/parallel/container_algorithms/rotate.hpp
+  hpx/parallel/container_algorithms/reverse.hpp
+  hpx/parallel/container_algorithms/replace.hpp
+  hpx/parallel/container_algorithms/remove.hpp
+  hpx/parallel/container_algorithms/remove_copy.hpp
+  hpx/parallel/container_algorithms.hpp
+  hpx/parallel/container_algorithms/partition.hpp
+  hpx/parallel/container_algorithms/move.hpp
+  hpx/parallel/container_algorithms/minmax.hpp
+  hpx/parallel/container_algorithms/merge.hpp
+  hpx/parallel/container_algorithms/is_heap.hpp
+  hpx/parallel/container_algorithms/generate.hpp
+  hpx/parallel/container_algorithms/for_each.hpp
+  hpx/parallel/container_algorithms/find.hpp
+  hpx/parallel/container_algorithms/fill.hpp
+  hpx/parallel/container_algorithms/count.hpp
+  hpx/parallel/container_algorithms/copy.hpp
+  hpx/parallel/container_algorithms/all_any_none.hpp
```

Comment:
Should we separate the container algorithms?

Reply:
I considered this, but since the container algorithms are more or less just overloads of the algorithms (correct me if I'm wrong) I thought they fit together. The other reason is that it's all headers so lumping them together doesn't really cost anything in compile time (this is not a good reason in general though...). I'm open to having them separate as well.